### DISCUSSION TITLE
Werte aus Eve Room o.ä ins KNX übertragen

### DISCUSSION THREAD

**Discussion by lucien-losweisser-de** [brutella/hkknx-public#259] [Q&A] [answered]
Hallo, ich möchte die Raumtemperatur mit Eve Room oder Vergleichbarem ermitteln und bei Änderung oder in Intervallen auf eine Gruppenadresse in meinem KNX-System übertragen ,um dann die Temperatur im Raum über KNX zu regeln. Hintergrund: Meine fest eingebauten Temperatur-Sensoren sind aus architektonischen Gründen an den Aussenwänden und liefern damit wenig sinnhafte Werte.

Meine naive Vorstellung ist, dass ich die Datenübertragung Eve --> KNX mit einem Skript lösen können sollte. Die Skript-Beispiele aus der Doku von hkknx übertragen die Daten in die umgekehrte Richtung. Hat jemand die Datenübertragung von Thread-Geräten wie Eve in Richtung KNX realisiert? Geht das überhaupt? Wäre das ein neues Feature? Bin gespannt auf Eure Antworten. Danke dafür. Lucien

**Comment by brutella** [comment, marked as answer]
> Hat jemand die Datenübertragung von Thread-Geräten wie Eve in Richtung KNX realisiert? Geht das überhaupt?

Technisch geht das. Aber auch nur theoretisch. Um Zugriff auf ein HomeKit Gerät zu bekommen, musst du dich mit diesem Gerät koppeln/pairen. Sobald ein Gerät gekoppelt ist, kann es kein zweites Mal gekoppelt werden. Wenn du es also zu HomeKit hinzufügst, kann man es kein zweites Mal hinzufügen, ohne es vorher zurückzusetzen. Somit musst du dich entscheiden: willst du es programmatisch auslesen oder in HomeKit haben. Beides geht so ohne weiteres nicht.

Weiters kenn ich kein Programm/Skript/Bibliothek mit der man sich mit einem HomeKit Gerät koppeln kann. Wie gesagt, theoretisch geht es aber es ist halt unpraktisch, da man das Gerät dann nicht mehr mit HomeKit verwenden kann.

**Comment by lucien-losweisser-de** [comment]
Vielen Dank Matthias für die superschnelle Antwort. Vorstellbar wäre ja dann, zumindest theoretisch, dass das HomeKit-Gerät nur mit der Anwendung nicht aber mit HomeKit gekoppelt wäre, diese Anwendung mit "GroupWrite" die Information an die entsprechende GA in KNX übergibt und man selbige Information aus dieser GA über hkknx wieder holt und im HomeKit darstellt, denke ich. Ich hoffe, ich bin verständlich. Dann ginge keine Information verloren, bzw. man hätte sie in beiden Welten verfügbar. Man müsste nur die Vorstellung in die Realität umsetzen ...

**Reply by chef76** [reply]
... oder du verwendest eine HomeKit Automation, welche bei Überschreitung einer Temperatur das Ventil schließt und bei Unterschreitung wieder öffnet. Ich Schalte so andere Sachen.

**Reply by brutella** [reply]
> Man müsste nur die Vorstellung in die Realität umsetzen ...

Eine Umsetzung ist in deinem Fall nicht so einfach möglich. Da die Eve Room-Geräte über Bluetooth kommunizieren, müsste hkknx auf einem Computer laufen, der eine Bluetooth Verbindung aufbauen kann. Da die Reichweite von Bluetooth begrenzt ist, müsste sich der Computer und das Eve-Geräte nahe beieinander befinden. Für das Koppeln und Entkoppeln müsste man eigene Benutzeroberflächen bauen, usw... Ist also alles nicht so einfach wie gedacht. 

Da wäre es einfacher, sich eine Temperatursensor von Shelly zu besorgen und diesen ins WLAN zu hängen. Dieser könnte dann in einem Skript über HTTP abgefragt werden.

**Reply by lucien-losweisser-de** [reply]
> Da wäre es einfacher, sich eine Temperatursensor von Shelly zu besorgen und diesen ins WLAN zu hängen. Dieser könnte dann in einem Skript über HTTP abgefragt werden.

Als Zwischenlösung, auch preislich, nicht uninteressant. Vielleicht ermöglicht Apple ja in irgendeiner Zukunft doch den Zugriff auf die Homekitgeräte und deren aktuelle Werte ... Da das länger dauern kann, danke für Deine Anregung.

**Comment by lucien-losweisser-de** [comment]
Danke für Deinen Rat. Der Punkt ist nur, dass ich das alles bereits im Homeserver realisiert habe und nicht wegschmeißen will. Im Prinzip will ich meine bestehende Welt behalten und um die HomeKit-Welt ergänzen. Also das Beste aus beiden Welten nehmen und verbinden.
